### PR TITLE
Added Mediaspeech ASR Dataset to egs2

### PR DESCRIPTION
**Pull request by abhesrivas**

### CODE REVIEW

**Comment by abhesrivas** [comment]
@sw005320 I think the linter test failed due to an issue with Zenodo and not specifically my PR. Could you please confirm this and if true, rerun the tests?

**Comment by sw005320** [comment]
I'll rerun the CI.

File: egs2/README.md [hunk 1]
```diff
@@ -96,3 +96,4 @@ See: https://espnet.github.io/espnet/espnet2_tutorial.html#recipes-using-espnet2
 | yesno                   | The "yesno" corpus                                                                      | ASR                     | HEB                  | http://www.openslr.org/1                                                                                     |              |
 | yoloxochitl_mixtec      | Yoloxochitl-Mixtec corpus (endangered language in central Mexico)                       | ASR                     | XTY                  | http://www.openslr.org/89                                                                                    |              |
 | zeroth_korean           | Zeroth-Korean                                                                           | ASR                     | KOR                  | http://www.openslr.org/40                                                                                    |              |
+mediaspeech                                                                           |   The ASR Corpus                   | ASR                  | 4 languages                                                                                     | http://www.openslr.org/108 |
```
Comment (sw005320) [inline egs2/README.md:99]: Can you list these four languages?

File: egs2/mediaspeech/asr1/RESULTS.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<!-- Generated by scripts/utils/show_asr_result.sh -->
```
Comment (sw005320) [inline egs2/mediaspeech/asr1/RESULTS.md:1]: - Can you merge this to README.md?
- Upload a model to the HF hub and put a link here.

File: egs2/mediaspeech/asr1/README.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+### Mediaspeech Corpus
+
+To run for a language other than Spanish, please set the `lang` flag in `local/data.sh` with the language tag (ES, TR, AR, FR).
```
Comment (sw005320) [inline egs2/mediaspeech/asr1/README.md:3]: Try to use a three-letter language code instead of a two-letter code.

**Review by sw005320 — COMMENTED** [review]
I have some other comments:

File: egs2/README.md (hunk 1 above)
Comment (abhesrivas) [reply egs2/README.md:99]: Done.

File: egs2/mediaspeech/asr1/README.md (hunk 3 above)
Comment (abhesrivas) [reply egs2/mediaspeech/asr1/README.md:3]: Actually, the zipped datasets are named after these two-letter codes (TR.zip, etc.) Therefore, I think including both could be less confusing. Or I can write: ES for Spanish, TR for Turkish, etc? Please let me know if you agree to this style.

Comment (sw005320) [reply egs2/mediaspeech/asr1/README.md:3]: oh, I see.
Your way is better then.
Sorry

File: egs2/mediaspeech/asr1/RESULTS.md (hunk 2 above)
Comment (abhesrivas) [reply egs2/mediaspeech/asr1/RESULTS.md]: Done.

**Comment by abhesrivas** [comment]
All proposed changes have been made. Would request you to review them again and let me know if any other changes are required. Thanks a lot!

**Review by sw005320 — APPROVED** [review]

**Comment by sw005320** [comment]
Can you resolve the conflict?

**Comment by abhesrivas** [comment]
Conflict is resolved. Thanks for the heads up!

**Comment by abhesrivas** [comment]
Hi Jiatong (@ftshijt), I have resolved the conflicts. Thanks for your time!

**Comment by simpleoier** [comment]
@abhesrivas Can you merge the latest ESPnet and reolve the fails in tests?

**Comment by abhesrivas** [comment]
Hi @simpleoier, I resolved the conflict. Please take a look, thanks!

**Comment by abhesrivas** [comment]
Hi, I fixed the conflict again. Could you please re-run the checks?

File: egs2/README.md [hunk 4]
```diff
@@ -59,46 +59,47 @@ See: https://espnet.github.io/espnet/espnet2_tutorial.html#recipes-using-espnet2
...
-| zeroth_korean           | Zeroth-Korean                                                                                                                    | ASR                     | KOR                  | http://www.openslr.org/40                                                                                    |              |
-| zh_openslr38            | ST-CMDS-20170001_1, Free ST Chinese Mandarin Corpus                                                                              | ASR                     | CMN                  | http://www.openslr.org/38                                                                                    |              |
\ No newline at end of file
+| mls                     | MLS (A large multilingual corpus derived from LibriVox audiobooks)                      | ASR                     | 8 languages          | http://www.openslr.org/94/                                                                                   |              |
```
Comment (ftshijt) [inline egs2/README.md:62]: Could you keep the other entries unchanged?

File: egs2/mediaspeech/asr1/local/data.sh [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+# -e 'error', -u 'undefined variable', -o ... 'error in pipeline', -x 'print commands',
+set -e
+set -u
+set -o pipefail
+
+. utils/parse_options.sh
+
+log "data preparation started"
+
+# mkdir -p ${JAVA}
```
Comment (ftshijt) [inline egs2/mediaspeech/asr1/local/data.sh:31]: Could you remove this commented line

File: egs2/mediaspeech/asr1/run.sh [hunk 6]
```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+# Set bash to 'debug' mode, it will exit on :
+# -e 'error', -u 'undefined variable', -o ... 'error in pipeline', -x 'print commands',
+set -e
+set -u
+set -o pipefail
+
+lid=false # whether to use language id as additional label
```
Comment (ftshijt) [inline egs2/mediaspeech/asr1/run.sh:8]: I do not see where you are using language id as an additional label, if not for this corpus, please remove it correspondingly.

File: egs2/mediaspeech/asr1/run.sh [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    --bpe_train_text "data/${train_set}/text" \
+    --feats_type raw \
+    --asr_config "${asr_config}" \
+    --feats_normalize uttmvn \
+    --inference_config "${inference_config}" \
+    --train_set "${train_set}" \
+    --valid_set "${train_dev}" \
+    --test_sets "${test_set}" \
+    --lm_train_text "data/${train_set}/text" \
+    --local_score_opts "--score_lang_id ${lid}" "$@"
```
Comment (ftshijt) [inline egs2/mediaspeech/asr1/run.sh:40]: Ditto

File: egs2/mediaspeech/asr1/local/data.sh [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+log "data preparation started"
+mkdir -p ${MEDIASPEECH}
+if [ -z "${MEDIASPEECH}" ]; then
+    log "Fill the value of 'MEDIASPEECH' of db.sh"
+    exit 1
+fi
+
+workspace=$PWD
+lang=ES
```
Comment (ftshijt) [inline egs2/mediaspeech/asr1/local/data.sh:40]: Is this the only language supported? If not, try to provide some options in `run.sh`

**Review by ftshijt — COMMENTED** [review]
Thanks for your contribution. Some comments as follows

**Comment by ftshijt** [comment]
Since there is already mediaspeech supported in egs2 and the PR hasn't been updated for quite a long time, I will close this PR.

**Pull request closed without merging**